app/lib/actions/candlepin/product/content_create.rb

```diff
@@ -10,13 +10,22 @@ class ContentCreate < Candlepin::Abstract
           param :content_url
           param :owner
           param :os_versions
+          param :repository_id
         end
 
         def run
+          content_url = input[:content_url]
+          # We must retrieve the repository in the run phase, so the latest Pulp version_href is
+          # already set. This is needed to retrieve the latest repository.deb_content_url_options!
+          repository = ::Katello::Repository.find(input[:repository_id])
```

Comment:
Can this query go inside the `if` statement below? Just to avoid any unnecessary lookups f or non deb repositories.

Reply:
If the `:type` parameter tells me whether we have deb content or not, then I suppose it can.

Reply:
Ah you have to look up the repo to tell the type, I see. Up to you if you want to refactor it so the repo type gets passed in as a parameter (unless it does already).

Reply:
Now that the basic functionality is complete I should have another look at implementing this suggestion. I will probably get to it early next week.

Reply:
Finally got around to implementing this suggestion.